sparse_ho/criterion.py

```diff
@@ -355,31 +334,20 @@ def __init__(self, X, y, model, sigma, finite_difference_step=None,
         self.val_test = None
         self.rmse = None
 
-    def v(self, mask, dense):
-        return (2 * self.X_val[:, mask].T @ (
-                self.X_val[:, mask] @ dense - self.y_val -
-                self.delta * self.sigma ** 2 / self.epsilon))
-
-    def v2(self, mask, dense):
-        return ((2 * self.sigma ** 2 *
-                 self.X_val[:, mask].T @ self.delta / self.epsilon))
-
-    def value(self, mask, dense, mask2, dense2):
-        dof = ((self.X_val[:, mask2] @ dense2 -
-                self.X_val[:, mask] @ dense) @ self.delta)
+    def value(self, X, y, mask, dense, mask2, dense2):
+        dof = ((X[:, mask2] @ dense2 -
+                X[:, mask] @ dense) @ self.delta)
         dof /= self.epsilon
         # compute the value of the sure
-        val = norm(self.y_val - self.X_val[:, mask] @ dense) ** 2
-        val -= self.X_val.shape[0] * self.sigma ** 2
+        val = norm(y - X[:, mask] @ dense) ** 2
+        val -= X.shape[0] * self.sigma ** 2
         val += 2 * self.sigma ** 2 * dof
 
         return val
 
-    def value_test(self, mask, dense):
-        # self.val_test = None
+    def value_test(self, X, y, mask, dense):
```

Comment:
to simplify I would remove all X_test, y_test, value_test in this PR, so that we see more clearly which functions do what